python/ray/dag/compiled_dag_node.py

```diff
@@ -1036,24 +1089,13 @@ def _get_or_compile(
                 self.actor_to_tasks[actor_handle].append(task)
             elif isinstance(task.dag_node, InputNode):
                 reader_and_node_list: List[Tuple["ray.actor.ActorHandle", str]] = []
-                # TODO (kevin85421): Currently, the shared memory channel doesn't
-                # support multiple readers on the same actor. However, if the
-                # InputNode is an actor instead of the driver process, we can support
-                # multiple readers on the same actor if the readers are on the same
-                # actor using IntraProcessChannel, which supports multiple readers
-                # on the same actor. We need to remove reader_handles_set in the future
-                # when we support multiple readers for both shared memory channel
-                # and IntraProcessChannel.
-                reader_handles_set = set()
                 for idx in task.downstream_actor_idxs:
                     reader_task = self.idx_to_task[idx]
                     assert isinstance(reader_task.dag_node, ClassMethodNode)
                     reader_handle = reader_task.dag_node._get_actor_handle()
-                    if reader_handle not in reader_handles_set:
-                        reader_and_node_list.append(
-                            (reader_handle, self._get_node_id(reader_handle))
-                        )
-                    reader_handles_set.add(reader_handle)
```

Comment:
why removing this?

Reply:
This dedups the readers which is incorrect when computing number of readers when we support multi-reads.